contrib/coreos/user-data.example

```diff
@@ -190,58 +209,67 @@ write_files:
+          if [ "$HOST" = "$PUBLIC_IP" ] ; then
+            OSD_ID=$(etcdctl get /deis/store/osds/$PUBLIC_IP)
             break
           fi
-          echo "waiting" && sleep 5
-          TIMEWAITED=$((TIMEWAITED+5))
         done
-        docker stop deis-store-daemon
-        docker exec deis-store-admin ceph osd crush remove osd.$OSD_ID
-        docker exec deis-store-admin ceph auth del osd.$OSD_ID
-        docker exec deis-store-admin ceph osd rm $OSD_ID
-        etcdctl rm /deis/store/osds/$PUBLIC_IP
-        etcdctl rm /deis/store/hosts/$PUBLIC_IP && sleep 10
-        # remove ceph mon
-        docker stop deis-store-monitor || true
-        docker exec deis-store-admin ceph mon remove `hostname -f` # fixme
-        docker stop deis-store-metadata || true
-      fi
+        # if we own an osd and its healthy, try to gracefully remove it
+        if [ ! -z "$OSD_ID" ] && [ ${#OSD_HOSTS[@]} -gt "3" ]; then
+          ADMIN_RUNNING=$(docker inspect --format="{{ .State.Running }}" deis-store-admin)
+          if [ $? -eq 1 ] || [ "$ADMIN_RUNNING" == "false" ]; then
+            echo "deis-store-admin container is required for graceful shutdown"
+            exit 2
+          fi
+          set -e -x -o pipefail
+          CURRENT_STATUS=$(docker exec deis-store-admin ceph health | awk '{print $1}')
+          if [[ "$CURRENT_STATUS" != *"HEALTH_OK"* ]]; then
+            echo "Ceph cluster must be healthy to perform graceful removal"
+            exit 3
+          fi
 
-      # removing the node from etcd
-      NODE=$(etcdctl member list | grep `cat /etc/machine-id` | cut -d ':' -f 1)
-      etcdctl member remove $NODE
+          docker exec deis-store-admin ceph osd out $OSD_ID
+          sleep 30
+          TIMEWAITED=0
+          until [[ $(docker exec deis-store-admin ceph health) == *"HEALTH_OK"* ]]
+          do
+            if [ $TIMEWAITED -gt "1200" ]
+            then
+              echo "ceph graceful removal timeout exceeded"
+              break
+            fi
+            echo "waiting" && sleep 5
+            TIMEWAITED=$((TIMEWAITED+5))
+          done
+          docker stop deis-store-daemon
+          docker exec deis-store-admin ceph osd crush remove osd.$OSD_ID
+          docker exec deis-store-admin ceph auth del osd.$OSD_ID
+          docker exec deis-store-admin ceph osd rm $OSD_ID
+          etcdctl rm /deis/store/osds/$PUBLIC_IP
+          etcdctl rm /deis/store/hosts/$PUBLIC_IP && sleep 10
+          # remove ceph mon
+          docker stop deis-store-monitor || true
+          docker exec deis-store-admin ceph mon remove `hostname -f` # fixme
+          docker stop deis-store-metadata || true
+        fi
+      }
+      etcd_shutdown () {
+        set -e -x -o pipefail
+        # removing the node from etcd
+        NODE=$(etcdctl member list | grep `cat /etc/machine-id` | cut -d ':' -f 1)
```

Comment:
I think setting ETCD_NAME would be outside the scope of this PR. I think it also might be deceptively tricky to implement. Since this system only exists to work with the current vanilla contrib/ setup, the current code should still work (even though its been a while). Does that seem reasonable?